datajunction-server/datajunction_server/database/user.py

```diff
@@ -0,0 +1,38 @@
+"""User database schema."""
+from typing import Optional
+
+from sqlalchemy import BigInteger, Integer
+from sqlalchemy.orm import Mapped, mapped_column
+
+from datajunction_server.database.connection import Base
+from datajunction_server.enum import StrEnum
+from datajunction_server.models.base import sqlalchemy_enum_with_name
+
+
+class OAuthProvider(StrEnum):
+    """
+    Support oauth providers
+    """
+
+    BASIC = "basic"
+    GITHUB = "github"
+    GOOGLE = "google"
+
+
+class User(Base):  # pylint: disable=too-few-public-methods
+    """Class for a user."""
+
+    __tablename__ = "users"
+
+    id: Mapped[int] = mapped_column(
+        BigInteger().with_variant(Integer, "sqlite"),
```

Comment:
Since we're exclusively using postgres now for the metadata db, do we need this with variant option for sqlite? Is that just for tests? Does it make sense to eventually have the tests use docker to start and init a new postgres instance?

Reply:
Yeah, we should eventually have the tests run with postgres and not on top of sqlite. This is setup is indeed just for the unit tests.